### DISCUSSION TITLE
Inference on a new dataset

### DISCUSSION THREAD

**Discussion by juampamuc** [MIC-DKFZ/nnUNet#1696] [Q&A] [answered]
Hello,

I work on two non-publicly available datasets. One I use for training. I was able to successfully train a model on it, and got really nice results. Congratulations for this very nice piece of software!
Now, I would like to evaluate this model on a second dataset. I have previously made sure that both have the same spacing. Now, my problem is that I cannot figure out how to tell nnUnet to use the configuration and plans from the first dataset on the second dataset.

As an example of the many wrong things I tried:

nnUNetv2_predict -d 2 -c 3d_fullres -p nnUNetPlans -i .\nnUNet_raw\Dataset002_whatever\imagesTr\ -o outfolder

Then it complains that it is not able to find the file nnUNet_results/Dataset002_whatever/nnUNetTrainer__nnUNetPlans__3d_fullres/dataset.json. But I have not generated any results for this dataset yet.

Can anyone please tell me what am I missing?

Regards

**Comment by ancestor-mithril** [comment]
If you want to use the model trained on dataset 1, you should use `-d 1` for predicting.

**Reply by juampamuc** [reply]
[ancestor-mithril](https://github.com/ancestor-mithril) Thank you for your reply. I am aware of that. The problem is how to use the model on a different dataset for which there are no plans defined. I need to make sure that I use the same pipeline for dataset 2 (data augmentation, postprocessing of outputs) as for dataset 1.

**Reply by ancestor-mithril** [reply]
If you want to do inference, you do not have to do anything special and just use `-d 1` during prediction. But if you want to finetune your model on dataset 2, see https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/pretraining_and_finetuning.md#pretraining-on-the-source-dataset.

**Reply by juampamuc** [reply]
Thank you again for pointing that out. That is the exact point. I forgot to mention that the second dataset is unlabelled, so when I run the commands as given in the documentation,

nnUNetv2_plan_and_preprocess -d 2

It complains that the labels are missing. So it seems that I am in a pickle.

**Reply by ancestor-mithril** [reply, marked as answer]
I don't think that nnUNet is able to do unsupervised pre-training. According to the documentation, it is able to do only supervised pretraining, so you need labels for the 2nd dataset, otherwise you can't perform any training on it.

**Reply by juampamuc** [reply]
Thank you very much for your time and prompt replies [ancestor-mithril](https://github.com/ancestor-mithril)
